### PR TITLE
Add fuzzy search

### PR DESCRIPTION
**Pull request by dongbohu**
This PR addresses issue #52.

### CODE REVIEW

**Comment by dongbohu** [comment]
See the demo site at: http://18.210.24.231/v1/nodes/?search=epi&fuzzy

File: dj_hetmech_app/views.py [hunk 1]
```diff
@@ -34,9 +34,6 @@ class NodeViewSet(ReadOnlyModelViewSet):
     http_method_names = ['get']
     serializer_class = NodeSerializer
     filter_backends = (filters.SearchFilter, )
-    # See the following page for "search" implementation and other filter options:
-    # https://www.django-rest-framework.org/api-guide/filtering/
-    search_fields = ('identifier', 'metanode__identifier', 'name')
```
Comment (dhimmel) [inline dj_hetmech_app/views.py:39]: I think we want to keep the ability to have an exact match on `identifier`. Possibly even a case insensitive exact match. We probably should give the exact identifier match the top score possible.

**Review by dhimmel — COMMENTED** [review]
Can you update the docstring to explain the search and the fuzzy parameter?

If easy to enable overriding trigram similarity threshold, may be helpful. Not a big deal if implementation is complex.

**Comment by dongbohu** [comment]
@dhimmel and @vincerubinetti: I changed `views.py` (see this PR) so that `search` now will show a union of the following nodes:
1. node `identifier` starts with the search string
2. node `name` is a substring of the search string
3. trigram similarity between node `name` and the search string is greater than `fuzzy` value (default to 0.3)

The results are sorted in this order. 

For example, this endpoint: 
http://18.210.24.231/v1/nodes/?search=epi
is a union of nodes that satisfies 1 and 2 (without fuzzy search).

This endpoint: 
http://18.210.24.231/v1/nodes/?search=epi&fuzzy
is a union of nodes that satisfies all three conditions, with fuzzy_value 0.3 (default).

This endpoint: 
http://18.210.24.231/v1/nodes/?search=epi&fuzzy=0.6
is a union of nodes that satisfies all three conditions, with fuzzy_value 0.6.

**Review by dhimmel — APPROVED** [review]
Much improved! Thanks @dongbohu.

**Comment by vincerubinetti** [comment]
This looks good to me. I don't know that we need to provide all of these options in the endpoint. For the front end, I'm probably just going to use `fuzzy=0.5` and leave it at that. Perhaps we just make the default `search=` automatically default to `fuzzy` on and `=0.5`.

**Comment by dongbohu** [comment]
It is very subtle to find a good default trigram similarity value. Both Django and Postgres documentations default this value to `0.3`, we probably should stay with it.

**Comment by dongbohu** [comment]
I set the fuzzy search as default so that this endpoint:
http://18.210.24.231/v1/nodes/?search=epi
is a union of nodes that satisfies 1, 2 and 3 with default  trigram similarity of 0.3.

This endpoint:
http://18.210.24.231/v1/nodes/?search=epi&similarity=0.6
is a union of nodes that satisfies 1, 2 and 3, with customized similarity value of 0.6.

This endpoint:
http://18.210.24.231/v1/nodes/?search=epi&similarity=1.0
is basically a union of nodes that satisfies 1 and 2, because `similarity=1.0` disables trigram search.

**Comment by dongbohu** [comment]
@dhimmel and @vincerubinetti: let me know if you have any other comments.

**Review by dhimmel — APPROVED** [review]
The new implementation with one parameter (`similarity`) is nice and simple

**Pull request merged**